File: Data Structures & Algorithms/can-place-flowers/submission-1.py
class Solution:
    def canPlaceFlowers(self, f: List[int], n: int) -> bool:
        count = 0 
        size = len(f)

        for i in range(size) :
            if f[i] == 0 :
                if i == 0 :
                    left = 0
                else :
                    left = f[i-1]
                
                if i == size - 1 :
                    right = 0
                else :
                    right = f[i+1]  

                if left == 0 and right == 0 :
                    f[i] = 1
                    count += 1     

            if count >= n :
                return True      

        return count >= n

File: Data Structures & Algorithms/binary-tree-diameter/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int diameterOfBinaryTree(TreeNode* root) {
        diameter = 0; // initialize before DFS
        dfs(root);
        return diameter;
    }

private:
    int diameter; // class member to track max diameter
    int dfs(TreeNode* node)
    {
        if(node == nullptr)
           return 0;

        int leftDepth = dfs(node->left);
        int rightDepth = dfs(node->right);

        diameter = max(diameter, leftDepth + rightDepth);

        return 1 + max(leftDepth,rightDepth);


    }
};
